include/fiction/algorithms/physical_design/ortho_ordering_network.hpp

```diff
@@ -0,0 +1,722 @@
+            // if any incoming edge is colored east, color them all east, and south otherwise
+            const auto color =
+                std::any_of(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                            [&ctn](const auto& fe) { return ctn.color_ntk.edge_color(fe) == ctn.color_east; }) ?
+                    ctn.color_east :
+                    ctn.color_south;
+
+            std::for_each(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                          [&ctn, &color](const auto& fe) { recursively_paint_edges(ctn, fe, color); });
+
+            // if all incoming edges are colored east, paint the node east as well
+            if (std::all_of(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                            [&ctn](const auto& fe) { return ctn.color_ntk.edge_color(fe) == ctn.color_east; }))
+            {
+                ctn.color_ntk.paint(mockturtle::node<Ntk>{n}, ctn.color_east);
+            }
+            // else, if all incoming edges are colored south, paint the node south as well
+            else if (std::all_of(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                                 [&ctn](const auto& fe) { return ctn.color_ntk.edge_color(fe) == ctn.color_south; }))
+            {
+                ctn.color_ntk.paint(mockturtle::node<Ntk>{n}, ctn.color_south);
+            }
+
+#if (PROGRESS_BARS)
+            // update progress
+            bar(i);
+#endif
+        });
+
+    // Adjust only the nodes viewed in the ordering network
+    ntk.foreach_pi(
+        [&](const auto& pi)
+        {
+            ntk.foreach_fanout(
+                pi,
+                [&](const auto& fon)
+                {
+                    // always track the current_node
+                    current_node = fon;
+
+                    // skip Inverters and paint them east
+                    if (ntk.is_inv(current_node))
+                    {
+                        // paint inverter east
+                        paint_node_and_edges(ctn, current_node, ctn.color_east);
+                        auto cur_fon = fanouts(ctn.color_ntk, current_node);
+                        // jump to next node
+                        current_node = cur_fon[0];
+                    }
+
+                    auto fo_one = ctn.color_ntk.get_fo_one();
+                    auto pi_pi  = ctn.color_ntk.get_pi_to_pi();
+
+                    if (ntk.is_fanout(current_node) && ntk.fanout_size(current_node) >= 2)
+                    {
+                        auto fo_two = ctn.color_ntk.get_fo_two();
+                        // paint the fan-out node east
+                        paint_node_and_edges(ctn, current_node, ctn.color_east);
+                        // 1. fan-out node related to two PIs
+                        if (const auto it = find(fo_two.cbegin(), fo_two.cend(), pi); it != fo_two.cend())
+                        {
+                            int i = static_cast<int>(it - fo_two.cbegin());
```

Comment:
Why not use
```suggestion
                            const auto i = it - fo_two.cbegin();
```

Reply:
Even better would probably be
```suggestion
                            auto i = std::distance(fo_two.cbegin(), it);
```
if that is what you're trying to do here.